src/smoothscroll.js

```diff
@@ -438,6 +444,15 @@ function polyfill() {
 if (typeof exports === 'object') {
   // commonjs
   module.exports = { polyfill: polyfill };
+
+  // avoids "ReferenceError: window is not defined" when running in node
+  if (typeof window === 'undefined') {
```

Comment:
Can't this be solved by lazy loading the script in the server like:

```js
if (typeof window !== 'undefined') {
  import('smoothscroll-polyfill')
    .then(module => module.polyfill())
}
```

Reply:
Unfortunately not. I'm in a crazy Angular-world. My code goes into a bundle via rollup.js (using [ng-packagr](https://github.com/dherges/ng-packagr)). That bundle goes again into a bigger bundle via webpack (using [angular-cli](https://github.com/angular/angular-cli])). Code must be tree-tree-shakable and everything. It's the exact opposite of lazy-loading. :laughing:

The code is already in production (dependency `"github:angular-schule/smoothscroll#eaa7009a7581216825f30c1a2a69cfe24f965fe5"`) and works fine in the client and and does nothing on the server. See [here](https://angular.schule/) and click `Seminarinhalte`. It nicely scrolls as always and also does not explode on the express server (prerendering).